source/features/insert-summary-button.tsx

```diff
@@ -0,0 +1,55 @@
+import React from 'dom-chef';
+import select from 'select-dom';
+import onetime from 'onetime';
+import delegate from 'delegate';
+import features from '../libs/features';
+import * as icons from '../libs/icons';
+
+const addEvents = onetime(() => {
+	delegate('.rgh-summary-btn', 'click', addSummaryDetails)
+});
```

Comment:
The toolbar is duplicated by GitHub when needed and sadly events aren't kept. This is also how we handle the upload button.

I tried implementing an idempotent `delegate` many times and failed because the code got unwieldy every time. Please someone do it.

Reply:
Hmm interesting 🤔 I could try to implement what you're going for, but I'm still learning things as I come across them, so I would have to play around with `delegate` to fully understand how it works first.

Reply:
What you should know is that the regular `Element#addEventListener` already de-duplicates event handlers, as long as _all_ of the parameters are identical:

```js
document.addEventListener('click', console.log); // ✅ added 
document.addEventListener('click', console.log); // ❌ deduplicated
document.addEventListener('click', console.log, false); // ✅ added, because there's a third parameter different from default
```

If we had native [multi-key WeakMaps](https://github.com/sindresorhus/module-requests/issues/108) it would be a piece of cake to implement, but without them you'd also need to write that functionality.

Reply:
Both [`delegate-it`](https://github.com/bfred-it/delegate-it) and [`many-keys-maps`](https://github.com/bfred-it/many-keys-map) have been published :)